sympy/solvers/inequalities.py

```diff
@@ -982,3 +987,457 @@ def reduce_inequalities(inequalities, symbols=[]):
+            # If there are no such columns, we have found a solution.
+            piv_cols = []
+            for j in range(C.cols):
+                if C[j] < 0:
+                    piv_cols.append(j)
+            if not piv_cols:
+                break
+
+            # From the columns being considered, we randomly pick one (to avoid cycling).
+            # This column, j0, will contain the pivot.
+            rand.shuffle(piv_cols)
+            j0 = piv_cols[0]
+
+            # Then we consider each row which has a positive entry at its intersection with the column we picked.
+            # If there are no such rows, the problem is unbounded.
+            piv_rows = [i for i in range(A.rows) if A[i, j0] > 0]
+            if not piv_rows:
+                raise UnboundedLinearProgrammingError('Objective function can assume arbitrarily large values!')
+
+        # For each row, we calculate a ratio: B[i] / A[i, j0].
+        # The row with the smallest ratio, i0, contains the pivot entry at its intersection with j0.
+        # If there are ties, pick a row at random.
+        min_ratio = float('inf')
+        min_rows = []
+        for i in piv_rows:
+            ratio = B[i] / A[i, j0]
+            if ratio < min_ratio:
+                min_ratio = ratio
+                min_rows = [i]
+            elif ratio == min_ratio:
+                min_rows.append(i)
+        rand.shuffle(min_rows)
+        i0 = min_rows[0]
+
+        M = _pivot(M, i0, j0)
+        R[j0], S[i0] = S[i0], R[j0]
+
+    argmax = [None]*(M.cols-1)
+    argmin_dual = [None]*(M.rows-1)
+
+    for i, var in enumerate(R):
+        v = var[0]
+        n = int(var[2])
+        if v == "x":
+            argmax[n] = 0
+        else:
+            argmin_dual[n] = M[-1, i]
+
+    for i, var in enumerate(S):
+        v = var[0]
+        n = int(var[2])
+        if v == "y":
+            argmin_dual[n] = 0
+        else:
+            argmax[n] = M[i, -1]
+
+    return M[-1, -1], argmax, argmin_dual
+
+
+def _to_standard_form(constraints, objective):
+    """
+    Converts a list of constraints and an objective function into the standard form for linear programming.
+    Each free variable is assigned a column lexicographically.
+
+    Examples
+    ========
+
+    >>> from sympy.solvers.inequalities import _to_standard_form
+    >>> from sympy.abc import x, y, z
+    >>> from sympy import Eq
+    >>> r1 = y+2*z <= 3
+    >>> r2 = -x-3*z <= -2
+    >>> r3 = 2*x+y+7*z <= 5
+    >>> A, B, C = _to_standard_form([r1, r2, r3], x + y + 5*z)
+    >>> A
+    Matrix([
+    [ 0, 1,  2],
+    [-1, 0, -3],
+    [ 2, 1,  7]])
+    >>> B
+    Matrix([
+    [ 3],
+    [-2],
+    [ 5]])
+    >>> C
+    Matrix([[1, 1, 5]])
+    >>> A, B, C = _to_standard_form([Eq(x, 3)], x*10)  # x = 3 become x >= 3 and x <= 3
+    >>> A
+    Matrix([
+    [ 1],
+    [-1]])
+    >>> B
+    Matrix([
+    [ 3],
+    [-3]])
+    >>> C
+    Matrix([[10]])
+    """
+    variables = objective.free_symbols
```

Comment:
I don't think that using free_symbols is a good idea. The caller needs to know the order of the symbols so they should be expected to provide that just like for `linear_eq_to_matrix`.

Reply:
They get sorted lexicographically in this line so the caller does know the order.
```
variables = sorted(variables, key=lambda v: str(v)) # order symbols lexicographically
```

But maybe it would be clearer if the order was provided?

Reply:
Why would they want a lexicographic order?

From the Zen of Python:

> In the face of ambiguity refuse the temptation to guess.

If an arbitrary order is needed then let the user provide it.